### PR TITLE
[styled-components] Replace usage of React's global JSX namespace with scoped JSX

### PR DESCRIPTION
**Pull request by eps1lon**
Was deprecated in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64464. Cherry-picked from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67588.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## styled-components ([<kbd>unpkg</kbd>](https://unpkg.com/browse/styled-components@latest/))
was missing the following properties:
1. StyleSheetConsumer
1. StyleSheetContext
1. version

## Formatting

The following files are not formatted:
1. types/styled-components/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a42733d00144d6115be8f57c38dcc7a6b429c573
</p>

**Review by navya9singh — APPROVED** [review]

File: types/styled-components/index.d.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ type ReactDefaultProps<C> = C extends { defaultProps: infer D } ? D : never;
 export type AnyStyledComponent = StyledComponent<any, any, any, any> | StyledComponent<any, any, any>;
 
 export type StyledComponent<
-    C extends keyof JSX.IntrinsicElements | React.ComponentType<any>,
+    C extends keyof React.JSX.IntrinsicElements | React.ComponentType<any>,
```
Comment (lifeiscontent) [inline types/styled-components/index.d.ts:127]:
```suggestion
    C extends React.ElementType,
```

File: types/styled-components/index.d.ts [hunk 2]
```diff
@@ -158,13 +158,13 @@ export interface StyledComponentBase<
         O & StyledComponentInnerOtherProps<WithC>,
         A | StyledComponentInnerAttrs<WithC>
     >;
-    withComponent<WithC extends keyof JSX.IntrinsicElements | React.ComponentType<any>>(
+    withComponent<WithC extends keyof React.JSX.IntrinsicElements | React.ComponentType<any>>(
```
Comment (lifeiscontent) [inline types/styled-components/index.d.ts:161]:
```suggestion
    withComponent<WithC extends keyof React.ElementType>(
```

File: types/styled-components/index.d.ts [hunk 3]
```diff
@@ -158,13 +158,13 @@ export interface StyledComponentBase<
...
         component: WithC,
     ): StyledComponent<WithC, T, O, A>;
 }
 
 export interface ThemedStyledFunctionBase<
-    C extends keyof JSX.IntrinsicElements | React.ComponentType<any>,
+    C extends keyof React.JSX.IntrinsicElements | React.ComponentType<any>,
```
Comment (lifeiscontent) [inline types/styled-components/index.d.ts:167]:
```suggestion
    C extends React.ElementType,
```

File: types/styled-components/index.d.ts (hunk 1 above)
Comment (lifeiscontent) [reply types/styled-components/index.d.ts:127]: @eps1lon wouldn't this be the same thing?

**Review by lifeiscontent — APPROVED** [review]

File: types/styled-components/index.d.ts (hunk 1 above)
Comment (eps1lon) [reply types/styled-components/index.d.ts:127]: I believe so, yes.

**Comment by eps1lon** [comment]
Ready to merge

**Pull request merged**